File: algorithms/searching/traversal/depth_first_search.py
def depth_first_search_pre_order(current_node, result=[]):
    result.append(current_node.value)

    if current_node.left:
        depth_first_search_pre_order(current_node.left, result)
    if current_node.right:
        depth_first_search_pre_order(current_node.right, result)

    return result


def depth_first_search_in_order(current_node, result=[]):
    if current_node.left:
        depth_first_search_in_order(current_node.left, result)

    result.append(current_node.value)

    if current_node.right:
        depth_first_search_in_order(current_node.right, result)

    return result


def depth_first_search_post_order(current_node, result=[]):
    if current_node.left:
        depth_first_search_post_order(current_node.left, result)
    if current_node.right:
        depth_first_search_post_order(current_node.right, result)

    result.append(current_node.value)

    return result

File: algorithms/searching/traversal/depth_first_search.js
function depthFirstSearchPreOrder(currentNode, result=[]) {
    result.push(currentNode.value);

    if (currentNode.left) {
        depthFirstSearchPreOrder(currentNode.left, result);
    }
    if (currentNode.right) {
        depthFirstSearchPreOrder(currentNode.right, result);
    }

    return result;
}


function depthFirstSearchInOrder(currentNode, result=[]) {
    if (currentNode.left) {
        depthFirstSearchInOrder(currentNode.left, result);
    }

    result.push(currentNode.value);

    if (currentNode.right) {
        depthFirstSearchInOrder(currentNode.right, result);
    }

    return result;
}


function depthFirstSearchPostOrder(currentNode, result=[]) {
    if (currentNode.left) {
        depthFirstSearchPostOrder(currentNode.left, result);
    }
    if (currentNode.right) {
        depthFirstSearchPostOrder(currentNode.right, result);
    }

    result.push(currentNode.value);

    return result;
}